src/Components/Form.js

```diff
@@ -13,77 +33,122 @@ class Form extends React.Component {
       name: "",
       email: "",
       message: "",
-      sent: false,
+      formErrors: {
+        name: "",
+        email: "",
+        message: "",
+      },
     };
   }
 
-  handleName = (e) => {
-    this.setState({
-      name: e.target.value,
-    });
-  };
-
-  handleEmail = (e) => {
-    this.setState({
-      email: e.target.value,
+  toastifySuccess() {
+    toast.success("Form sent!", {
+      position: "bottom-right",
+      autoClose: 5000,
+      hideProgressBar: true,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: false,
     });
-  };
+  }
 
-  handleMessage = (e) => {
-    this.setState({
-      message: e.target.value,
+  toastifyFail() {
+    toast.error("Form failed to send! Did you fill out all the fields?", {
+      position: "bottom-right",
+      autoClose: 5000,
+      hideProgressBar: true,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: false,
     });
-  };
+  }
 
-  formSubmit = (e) => {
+  handleSubmit = (e) => {
     e.preventDefault();
 
-    let data = {
-      name: this.state.name,
-      email: this.state.email,
-      message: this.state.message,
-    };
-
-    Axios.post("/api/forma", data)
-      .then((res) => {
-        this.setState(
-          {
-            sent: true,
-          },
-          this.resetForm()
-        );
-      })
-      .catch(() => {
-        console.log("Message not sent");
-      });
+    if (formValid(this.state)) {
+      const { name, email, message } = this.state;
+
+      let templateParams = {
+        name: name,
+        email: email,
+        message: message,
+      };
+
+      emailjs.send(
+        "gmail",
+        "portfolio_contact",
+        templateParams,
+        "user_uFBt7OJSbpoxTLb0kocyH"
+      );
+
+      console.log(`
+      ---SUBMITTING---
+      Name: ${name}
+      Email: ${email}
+      Message: ${message}
+      `);
+
+      this.toastifySuccess();
+      this.resetForm();
+    } else {
+      console.error("FORM INVALID - DISPLAY ERROR MESSAGE");
+      this.toastifyFail();
+    }
   };
 
-  resetForm = () => {
+  resetForm() {
     this.setState({
       name: "",
       email: "",
       message: "",
     });
+  }
 
-    setTimeout(() => {
-      this.setState({
-        sent: false,
-      });
-    }, 3000);
+  handleChange = (e) => {
+    e.preventDefault();
+    const { name, value } = e.target;
+    let formErrors = { ...this.state.formErrors };
+
+    switch (name) {
```

Comment:
would be great to decouple the `handleChange` function and pull this switch statement out into another function. Something like `handleFormErrors()`